packages/lesswrong/components/posts/PostsPreviewTooltip.tsx

```diff
@@ -139,20 +140,31 @@ const getPostCategory = (post: PostsBase) => {
     return null;
 }
 
-const PostsPreviewTooltip = ({ postsList, post, classes, comment }: {
+const PostsPreviewTooltip = ({ postsList, post, hash, classes, comment }: {
   postsList?: boolean,
+  hash?: string|null,
   post: PostsList|SunshinePostsList|null,
   classes: ClassesType,
   comment?: any,
 }) => {
-  const { PostsUserAndCoauthors, PostsTitle, ContentItemBody, CommentsNode, BookmarkButton, LWTooltip, FormatDate } = Components
+  const { PostsUserAndCoauthors, PostsTitle, ContentItemBody, CommentsNode, BookmarkButton, LWTooltip, FormatDate, Loading } = Components
   const [expanded, setExpanded] = useState(false)
 
-  if (!post) return null
+  const {document: postWithHighlight, loading} = useSingle({
+    collectionName: "Posts",
+    fragmentName: "HighlightWithHash",
+    documentId: post?._id,
+    skip: !post || !hash,
```

Comment:
If I understand this correctly, this now means all hover-previews for posts have an initial loading screen. I don't think this is worth the tradeoff. I feel like we should just keep using the `post` element that is passed if we are not being passed a `hash`, which skips the loading, and only load the postWithHighlight if we have indeed been passed a `hash` and that way we don't have any regression in user experience.

Reply:
You misunderstood; this won't have a loading spinner if there's no `hash` (I just tested).

Reply:
Cool, then all seems good. Somewhat surprised by this, but will check it out in prod.